gui/src/Users/reducers/users.reducer.ts

```diff
@@ -110,11 +110,16 @@ export const usersReducer = (state: any = initialState, action: any) => {
 
       let toAppend: any = {roles};
       if (!(state as any).selectedRoles) {
-        let selectedRoles = new Array(roles.length);
+        let selectedRoles = new Array(roles.length + 1);
 
-        for (let i = 0; i < selectedRoles.length; i++) {
+        selectedRoles[0] = {
```

Comment:
This doesn't seem like logic that should reside no the frontend. Let's get with @vitreuz and see if we can offload this to the backend.